### PR TITLE
Move local postgres DB installation to developer folder

### PR DESCRIPTION
**Pull request by Mo3m3n**

### CODE REVIEW

File: developer/openshift/dev_setup.sh [hunk 1]
```diff
@@ -275,6 +275,9 @@ install_pipeline_service() {
     --kustomization "git::$GIT_URL/developer/openshift/gitops/argocd?ref=$GIT_REF" |
     indent 2
 
+  echo "- Installing local postgres DB for tekton results:"
```
Comment (Roming22) [inline developer/openshift/dev_setup.sh:278]: Why isn't that a kustomization applied to pipeline-service? or minio?

File: operator/images/access-setup/content/bin/setup_work_dir.sh [hunk 2]
```diff
@@ -187,12 +187,14 @@ tekton_results_manifest(){
...
-      --from-literal=DATABASE_PASSWORD="$TEKTON_RESULTS_DATABASE_PASSWORD" \
+      --from-literal=db.user="$TEKTON_RESULTS_DATABASE_USER" \
+      --from-literal=db.password="$TEKTON_RESULTS_DATABASE_PASSWORD" \
+      --from-literal=db.host="tekton-results-database-service.tekton-results.svc.cluster.local" \
+      --from-literal=db.name="tekton_results" \
       --dry-run=client -o yaml > "$results_secret"
 
     yq e -n '.resources += ["namespace.yaml", "tekton-results-secret.yaml"]' > "$results_kustomize"
-    if [ "$(yq ".data" < "$results_secret" | grep -cE "DATABASE_USER|DATABASE_PASSWORD")" != "2" ]; then
+    if [ "$(yq ".data" < "$results_secret" | grep -cE "db.user|db.password")" != "2" ]; then
```
Comment (Roming22) [inline operator/images/access-setup/content/bin/setup_work_dir.sh:197]: You need to check for the additional data `db.host|db.name` as well.

File: operator/test/test.sh [hunk 3]
```diff
@@ -72,7 +72,7 @@ init() {
     pwd
   )
   export KUBECONFIG
-  NAMESPACE="pipeline-service-test"
+  NAMESPACE="plnsvc-tests"
```
Comment (Roming22) [inline operator/test/test.sh:75]: Curious, why the change of namespace?

File: operator/test/test.sh [hunk 4]
```diff
@@ -182,14 +182,14 @@ test_results() {
   test_pipelines
   echo -n "  - Results in database: "
 
-  # Prepare a custom Service Account that will be used for debugging purposes
-  if ! kubectl get serviceaccount tekton-results-debug -n tekton-results >/dev/null 2>&1; then
-    kubectl create serviceaccount tekton-results-debug -n tekton-results
+  # Service Account to test tekton-results
+  if ! kubectl get serviceaccount tekton-results-tests -n "$NAMESPACE" >/dev/null 2>&1; then
```
Comment (Roming22) [inline operator/test/test.sh:186]: The account name is used a lot. Maybe use a variable?

File: operator/test/test.sh [hunk 5]
```diff
@@ -210,7 +210,7 @@ test_results() {
     "call"
     "--channel_creds_type=ssl"
     "--ssl_target=tekton-results-api-service.tekton-results.svc.cluster.local"
-    "--call_creds=access_token=$(kubectl get secrets -n tekton-results -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='tekton-results-debug')].data.token}" | cut -d ' ' -f 2 | base64 --decode)"
+    "--call_creds=access_token=$(kubectl get secrets -n "$NAMESPACE" -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='tekton-results-tests')].data.token}"| cut -d ' ' -f 2 | base64 --decode)"
```
Comment (Roming22) [inline operator/test/test.sh:213]: `token}"| cut` -> `token}" | cut`

**Review by Roming22 — COMMENTED** [review]
Please rebase. If it makes sense, please squash the commits to make the history easier to navigate.

**Comment by Mo3m3n** [comment]
@Roming22 I have rebased the branch on last main.
 It is better to keep two seperate commits as we deal with two different subjects (local db creation, updating results test)

File: operator/test/test.sh (hunk 5 above)
Comment (Mo3m3n) [reply operator/test/test.sh]: Done

File: operator/test/test.sh (hunk 4 above)
Comment (Mo3m3n) [reply operator/test/test.sh]: Done

File: operator/test/test.sh (hunk 3 above)
Comment (Mo3m3n) [reply operator/test/test.sh:75]: the change in question was done in this branch before "pipeline-service-test" was set main.
And access in MVP infra is set for `plnsvc-tests`  so it is easier to put here the corresponding ns rather than ask to change the access for the pipeline-service test ns but it is possible if there are good reasons to keep the other NS name.

File: developer/openshift/dev_setup.sh (hunk 1 above)
Comment (Mo3m3n) [reply developer/openshift/dev_setup.sh:278]: I thought about this not as "kustomizing" an existing component (database) but as installing a new one as a dependency of pipeline-service, so the idea is to be explicit about its installation.
Maybe it is more readable to have it as kustomization of "pipeline-service" ? I can work on it

Comment (Mo3m3n) [reply developer/openshift/dev_setup.sh:278]: After having a closer look, trying a kustomize with pipeline-service will be messed by
https://github.com/openshift-pipelines/pipeline-service/blob/fe7507dafaa8f77bc5d1fafe2764df1c66a43d07/developer/openshift/dev_setup.sh#L286
Maybe we need to have both (the db and current-branch related modifs )in a kustomize in a subsequent PR

File: operator/test/test.sh [hunk 6]
```diff
@@ -210,7 +211,7 @@ test_results() {
     "call"
     "--channel_creds_type=ssl"
     "--ssl_target=tekton-results-api-service.tekton-results.svc.cluster.local"
-    "--call_creds=access_token=$(kubectl get secrets -n tekton-results -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='tekton-results-debug')].data.token}" | cut -d ' ' -f 2 | base64 --decode)"
+    "--call_creds=access_token=$(kubectl get secrets -n "$NAMESPACE" -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']==\"$RESULTS_SA\")].data.token}" | cut -d ' ' -f 2 | base64 --decode)"
```
Comment (avinal) [inline operator/test/test.sh:214]: We have recently changed how we are obtaining tokens, this method may or may not work in future interactions. 
```suggestion
    "--call_creds=access_token=$(kubectl create token $RESULTS_SA -n $NAMESPACE)"
```

**Review by avinal — CHANGES_REQUESTED** [review]
https://github.com/tektoncd/results/blob/main/config/api.yaml#L44-L49 if we do not use the same variable name, we have patch it again.

File: developer/openshift/dev_setup.sh (hunk 1 above)
Comment (Roming22) [reply developer/openshift/dev_setup.sh:278]: Fine, we can clean it up later if needed.

File: operator/gitops/argocd/pipeline-service/tekton-results/base/deployment.yaml [hunk 7]
```diff
@@ -24,36 +24,30 @@ spec:
...
-                --namespace="tekton-results" \
-                --from-literal=DATABASE_USER=tekton \
-                --from-literal=DATABASE_PASSWORD=$(dd if=/dev/urandom bs=20 count=1 2>/dev/null | base64)
       containers:
         - env:
             - name: DB_USER
               valueFrom:
                 secretKeyRef:
-                  key: DATABASE_USER
+                  key: db.user
```
[the remark this answers is not in the document]
Comment (Mo3m3n) [reply operator/gitops/argocd/pipeline-service/tekton-results/base/deployment.yaml:32]: This work is based on a currently working environment (with the current version of tekton-results) where the key is db.user.
If a different version of tekton-results is expecting `POSTGRES_USER`  that should be handled in the PR updating the tekton-results version

File: operator/gitops/argocd/pipeline-service/tekton-results/base/deployment.yaml [hunk 8]
```diff
@@ -24,36 +24,30 @@ spec:
...
                   name: tekton-results-database
             - name: DB_PASSWORD
               valueFrom:
                 secretKeyRef:
-                  key: DATABASE_PASSWORD
+                  key: db.password
```
[the remark this answers is not in the document]
Comment (Mo3m3n) [reply operator/gitops/argocd/pipeline-service/tekton-results/base/deployment.yaml:37]: answered above

File: operator/images/access-setup/content/bin/setup_work_dir.sh [hunk 9]
```diff
@@ -187,12 +187,14 @@ tekton_results_manifest(){
 
     kubectl create namespace tekton-results --dry-run=client -o yaml > "$results_namespace"
     kubectl create secret generic -n tekton-results tekton-results-database \
-      --from-literal=DATABASE_USER="$TEKTON_RESULTS_DATABASE_USER" \
-      --from-literal=DATABASE_PASSWORD="$TEKTON_RESULTS_DATABASE_PASSWORD" \
+      --from-literal=db.user="$TEKTON_RESULTS_DATABASE_USER" \
+      --from-literal=db.password="$TEKTON_RESULTS_DATABASE_PASSWORD" \
```
[the remark this answers is not in the document]
Comment (Mo3m3n) [reply operator/images/access-setup/content/bin/setup_work_dir.sh:191]: answered above

File: operator/images/access-setup/content/bin/setup_work_dir.sh [hunk 10]
```diff
@@ -187,12 +187,14 @@ tekton_results_manifest(){
...
+      --from-literal=db.host="tekton-results-database-service.tekton-results.svc.cluster.local" \
+      --from-literal=db.name="tekton_results" \
       --dry-run=client -o yaml > "$results_secret"
 
     yq e -n '.resources += ["namespace.yaml", "tekton-results-secret.yaml"]' > "$results_kustomize"
-    if [ "$(yq ".data" < "$results_secret" | grep -cE "DATABASE_USER|DATABASE_PASSWORD")" != "2" ]; then
+    if [ "$(yq ".data" < "$results_secret" | grep -cE "db.host|db.name|db.user|db.password")" != "4" ]; then
```
[the remark this answers is not in the document]
Comment (Mo3m3n) [reply operator/images/access-setup/content/bin/setup_work_dir.sh:197]: answered above

**Review by Roming22 — APPROVED** [review]
